test/spec/modules/schain_spec.js

```diff
@@ -0,0 +1,248 @@
+    expect(isSchainObjectValid(schainConfig)).to.false;
+    delete schainConfig.nodes[1].rid; // undefined // param is optional thus this will result true
+    expect(isSchainObjectValid(schainConfig)).to.true;
+    schainConfig.nodes[1].rid = true; // boolean
+    expect(isSchainObjectValid(schainConfig)).to.false;
+    schainConfig.nodes[1].rid = []; // array
+    expect(isSchainObjectValid(schainConfig)).to.false;
+  });
+
+  it('Returns false if nodes[].name is not a String', function() {
+    schainConfig.nodes[0].name = 'name value'; // string
+    expect(isSchainObjectValid(schainConfig)).to.true;
+    schainConfig.nodes[0].name = 1; // Integer
+    expect(isSchainObjectValid(schainConfig)).to.false;
+    schainConfig.nodes[0].name = 1.1; // float
+    expect(isSchainObjectValid(schainConfig)).to.false;
+    schainConfig.nodes[0].name = {}; // object
+    expect(isSchainObjectValid(schainConfig)).to.false;
+    delete schainConfig.nodes[0].name; // undefined // param is optional thus this will result true
+    expect(isSchainObjectValid(schainConfig)).to.true;
+    schainConfig.nodes[0].name = true; // boolean
+    expect(isSchainObjectValid(schainConfig)).to.false;
+    schainConfig.nodes[0].name = []; // array
+    expect(isSchainObjectValid(schainConfig)).to.false;
+  });
+
+  it('Returns false if nodes[].domain is not a String', function() {
+    schainConfig.nodes[1].domain = 'domain value'; // string
+    expect(isSchainObjectValid(schainConfig)).to.true;
+    schainConfig.nodes[1].domain = 1; // Integer
+    expect(isSchainObjectValid(schainConfig)).to.false;
+    schainConfig.nodes[1].domain = 1.1; // float
+    expect(isSchainObjectValid(schainConfig)).to.false;
+    schainConfig.nodes[1].domain = {}; // object
+    expect(isSchainObjectValid(schainConfig)).to.false;
+    delete schainConfig.nodes[1].domain; // undefined // param is optional thus this will result true
+    expect(isSchainObjectValid(schainConfig)).to.true;
+    schainConfig.nodes[1].domain = true; // boolean
+    expect(isSchainObjectValid(schainConfig)).to.false;
+    schainConfig.nodes[1].domain = []; // array
+    expect(isSchainObjectValid(schainConfig)).to.false;
+  });
+
+  it('Returns false if nodes[].ext param is not an Object', function() {
+    schainConfig.nodes[0].ext = 1; // Integer
+    expect(isSchainObjectValid(schainConfig)).to.false;
+    schainConfig.nodes[0].ext = 1.1; // float
+    expect(isSchainObjectValid(schainConfig)).to.false;
+    schainConfig.nodes[0].ext = {}; // object
+    expect(isSchainObjectValid(schainConfig)).to.true;
+    delete schainConfig.nodes[0].ext; // undefined // param is optional thus this will result true
+    expect(isSchainObjectValid(schainConfig)).to.true;
+    schainConfig.nodes[0].ext = true; // boolean
+    expect(isSchainObjectValid(schainConfig)).to.false;
+    schainConfig.nodes[0].ext = []; // array
+    expect(isSchainObjectValid(schainConfig)).to.false;
+  });
+});
+
+describe('', function() {
```

Comment:
Try to use something descriptive rather than empty quotes

Reply:
ohh my bad, i will fix it.